generator/config/stage/out.yaml

```diff
@@ -3,26 +3,38 @@ name: $out
 link: 'https://www.mongodb.com/docs/manual/reference/operator/aggregation/out/'
 type:
     - stage
-encode: object
+encode: single
```

Comment:
I have to change this type, because an object with only `{ coll }` is not accepted. The value must be a string or an object with `{ coll, db }`

We could have a factory that return a string or an object, that could be used in `$merge` also.

Reply:
@GromNaN: This reminds me of another operator where you were always encoding the long form (array I think) instead of a conditional short form. Perhaps that was `$type`? Although I'm not sure if it's worth addressing that since it isn't technically required as you noted here.

Reply:
In another [comment](https://github.com/mongodb/mongo-php-builder/pull/56/files#r1467468501), @alcaeus asked about creating custom encoder logic. Is that feasible with the current implementation, or is it more complicated because the entire OutStage file is generated (and thus would require addressing this through the generator code itself).

IIRC, the original intention of this project was to have an ongoing dependency on the generator, so I understand if there's presently no way to maintain stages and operators through PHP along (in an alternate reality where the generator was only used to create the initial implementation).

Asking the above more for my own curiosity.

I understand the current change to use `encode: single` here is the most flexible and probably sets you up to later use `string|OutCollection` for the parameter.